tools/integration_tests/operations/copy_dir_test.go

```diff
@@ -220,3 +219,147 @@ func TestCopyDirectoryInNonEmptyDirectory(t *testing.T) {
 	destSrc := path.Join(destDir, SrcCopyDirectory)
 	checkIfCopiedDirectoryHasCorrectData(destSrc, t)
 }
+
+func checkIfCopiedEmptyDirectoryHasCorrectData(destSrc string, t *testing.T) {
+	objs, err := os.ReadDir(destSrc)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(objs) != 0 {
+		t.Errorf("Directory has incorrect data.")
+	}
+}
+
+// Copy SrcDirectory in DestDirectory
+// emptySrcDirectoryCopyTest
+
+// destNonEmptyCopyDirectory
+// destNonEmptyCopyDirectory/subDirInNonEmptyDestCopyDirectory
+
+// Output
+// destNonEmptyCopyDirectory
+// destNonEmptyCopyDirectory/subDirInNonEmptyDestCopyDirectory
+// destNonEmptyCopyDirectory/emptySrcDirectoryCopyTest
+func TestCopyEmptyDirectoryInNonEmptyDirectory(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	srcDir := path.Join(setup.MntDir(), EmptySrcDirectoryCopyTest)
+	operations.CreateDirectoryWithNFiles(0, srcDir, "", t)
+
+	// Create below directory
+	// destNonEmptyCopyDirectory                                                -- Dir
+	// destNonEmptyCopyDirectory/subDirInNonEmptyDestCopyDirectory              -- Dir
+	destDir := path.Join(setup.MntDir(), DestNonEmptyCopyDirectory)
+	createDestNonEmptyDirectory(t)
+
+	err := operations.CopyDir(srcDir, destDir)
+	if err != nil {
+		t.Errorf("Error in copying directory: %v", err)
+	}
+
+	objs, err := os.ReadDir(destDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Check if destCopyDirectory has the correct directory copied.
+	// destNonEmptyCopyDirectory
+	// destNonEmptyCopyDirectory/emptyDirectoryCopyTest           - Dir
+	// destNonEmptyCopyDirectory/subDestCopyDirectory             - Dir
+	if len(objs) != NumberOfObjectsInDestCopyDirectory {
```

Comment:
Yes, we need to change this variable name also `NumberOfObjectsInDestCopyDirectory` -> `NumberOfObjectsInNonEmptyDestCopyDirectory`